src/code/views/teacher-view.tsx

```diff
@@ -147,8 +153,16 @@ export class TeacherView extends React.Component<
 
   renderGridMap() {
     const simulation = simulationStore.selected,
-          grid = simulation && simulation.grid;
+          grid = simulation && simulation.grid,
+          groupMap: { [index:string]: string } = {};
 
+    if (simulation) {
+      simulation.presences.presenceList.forEach((presence) => {
+        if (presence.groupName && presence.weatherStationID) {
+          groupMap[presence.weatherStationID] = presence.groupName;
+        }
+      });
+    }
     const colorFunc = (cell:IGridCell) => {
```

Comment:
Once upon a time `colorFunc` only computed the color of cells. Its been on my mind to rename it someday.  ;)

Reply:
Seems beyond the scope of this PR, so I"ll save it for another time.